src/common/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -640,6 +646,97 @@ class ngraph::pass::mask_propagation::Reduce : public MatcherPass {
     }
 };
 
+class ngraph::pass::mask_propagation::Reshape : public MatcherPass {
+public:
+    Reshape() {
+        auto inputs = pattern::any_input(pattern::has_static_shape());
+        // Can't process subgraphs as shape input as this value
+        // could demand to be adjusted at several dimensions.
+        // So, we can't just put -1 on one of the dims of subgraph.
+        auto weights = pattern::any_input();
+        auto reshape = pattern::wrap_type<opset6::Reshape>({inputs, weights});
+
+        ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+            const auto & pattern_map = m.get_pattern_value_map();
+            const auto m_weights = pattern_map.at(weights);
+            const auto & m_input = pattern_map.at(inputs);
+            const auto & m_output = pattern_map.at(reshape);
+
+            // Check if this reshape is before group convolution
+            // In such case this reshape should be processed by GroupConvolutionReshape pass
+            for (const auto inp : m_output.get_target_inputs())
+                if (is_type<opset6::GroupConvolution>(inp.get_node()))
+                    return true;
+
+            const auto constant = std::dynamic_pointer_cast<opset6::Constant>(m_weights.get_node_shared_ptr());
+            if (!constant) {
+                    NGRAPH_DEBUG << "Can't process rehape node " << m_weights.get_node()->get_friendly_name()
+                                 << " with no constant as shape input.";
+                    return false;
+            }
+
+            // Check reshape operation reshape only dimension without masks
+            if (auto input_mask = getMask(m_input)) {
+                auto output_mask = std::make_shared<Mask>(m_output.get_partial_shape().rank().get_length());
+                auto weights_mask = std::make_shared<Mask>(m_output.get_partial_shape().rank().get_length(), true);
+
+                const auto input_shape = m_input.get_shape();
+                const auto output_shape = constant->cast_vector<size_t>();
```

Comment:
Values in Reshape constant can be negative.
Please cast to int64_t

Reply:
Done